pkg/restore/restore_test.go

```diff
@@ -40,36 +40,39 @@ import (
 )
 
 func TestPrioritizeResources(t *testing.T) {
-	mapper := &FakeMapper{AutoReturnResource: true}
-	priorities := []string{"namespaces", "configmaps", "pods"}
-	resources := []*metav1.APIResourceList{
-		{
-			GroupVersion: "v1",
-			APIResources: []metav1.APIResource{
-				{Name: "aaa"},
-				{Name: "bbb"},
-				{Name: "configmaps"},
-				{Name: "ddd"},
-				{Name: "namespaces"},
-				{Name: "ooo"},
-				{Name: "pods"},
-				{Name: "sss"},
+	helper := &FakeDiscoveryHelper{
+		RESTMapper: &FakeMapper{AutoReturnResource: true},
+		ResourceList: []*metav1.APIResourceList{
+			{
+				GroupVersion: "v1",
+				APIResources: []metav1.APIResource{
+					{Name: "aaa"},
+					{Name: "bbb"},
+					{Name: "configmaps"},
+					{Name: "ddd"},
+					{Name: "namespaces"},
+					{Name: "ooo"},
+					{Name: "pods"},
+					{Name: "sss"},
+				},
 			},
 		},
 	}
+	var (
+		priorities       = []string{"namespaces", "configmaps", "pods"}
+		includesExcludes = collections.NewIncludesExcludes().Includes("*").Excludes("ooo")
+	)
 
-	result, err := prioritizeResources(mapper, priorities, resources)
+	result, err := prioritizeResources(helper, priorities, includesExcludes)
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
 
-	expected := []string{"namespaces", "configmaps", "pods", "aaa", "bbb", "ddd", "ooo", "sss"}
-	for i := range result {
-		if len(expected) < i+1 {
-			t.Errorf("result is too small: %v", result)
-			break
-		}
+	expected := []string{"namespaces", "configmaps", "pods", "aaa", "bbb", "ddd", "sss"}
+
+	assert.Equal(t, len(expected), len(result))
```

Comment:
Do you want this to be `require.Equal` so the loop below doesn't panic if the index is out of range?

Reply:
good idea, switched.